scipy/stats/_correlation.py

```diff
@@ -0,0 +1,200 @@
+        Default is 'asymptotic'. The following options are available.
+
+        * ``'asymptotic'``: compares the standardized test statistic
+          against the normal distribution.
+        * `PermutationMethod` instance. In this case, the p-value
+          is computed using `permutation_test` with the provided
+          configuration options and other appropriate settings.
+
+    y_continuous : bool, default: False
+        Whether `y` is assumed to be drawn from a continuous distribution.
+        If `y` is drawn from a continuous distribution, results are valid
+        whether this is assumed or not, but enabling this assumption will
+        result in faster computation and typically produce similar results.
+
+    Returns
+    -------
+    res : SignificanceResult
+        An object containing attributes:
+
+        statistic : float
+            The xi correlation statistic.
+        pvalue : float
+            The associated *p*-value: the probability of a statistic at least as
+            high as the observed value under the null hypothesis of independence.
+
+    See Also
+    --------
+    scipy.stats.pearsonr, scipy.stats.spearmanr, scipy.stats.kendalltau
+
+    References
+    ----------
+    .. [1] Chatterjee, Sourav. "A new coefficient of correlation." Journal of
+           the American Statistical Association 116.536 (2021): 2009-2022.
+           :doi:`10.1080/01621459.2020.1758115`.
+
+    Examples
+    --------
+    Generate perfectly correlated data, and observe that the xi correlation is
+    nearly 1.0.
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> rng = np.random.default_rng(348932549825235)
+    >>> x = rng.uniform(0, 10, size=100)
+    >>> y = np.sin(x)
+    >>> res = stats.xi_correlation(x, y)
+    >>> res.statistic
+    np.float64(0.9012901290129013)
+
+    The probability of observing such a high value of the statistic under the
+    null hypothesis of independence is very low.
+
+    >>> res.pvalue
+    np.float64(2.2206974648177804e-46)
+
+    As noise is introduced, the correlation coefficient decreases.
+
+    >>> noise = rng.normal(scale=[[0.1], [0.5], [1]], size=(3, 100))
+    >>> res = stats.xi_correlation(x, y + noise, axis=-1)
+    >>> res.statistic
+    array([0.79507951, 0.41824182, 0.16651665])
+
+    Because the distribution of `y` is continuous, continuity may be
+    assumed with little effect on the results.
+
+    >>> stats.xi_correlation(x, y + noise, y_continuous=True, axis=-1).statistic
+    array([0.79507951, 0.41824182, 0.16651665])
+
+    """
+    # x, y, `axis` input validation taken care of by decorator
+    # In fact, `axis` is guaranteed to be -1
+    y_continuous, method = _xi_correlation_iv(y_continuous, method)
+
+    # A highly negative statistic is possible, e.g.
+    # x = np.arange(100.), y = (x % 2 == 0)
+    # Unclear whether we should expose `alternative`, though.
+    alternative = 'greater'
```

Comment:
Theorem 1.1 says the statistic converges (in large sample) to a constant between 0 and 1. This does *not* require the null hypothesis but only requires the samples to contain independent observations. Remark (9) following Theorem 1.1 further elaborates this and mentions that negative xi_correlation, bar noise, can only indicate non-independent observation.

Therefore I’m inclined toward a “two-way” p-value, though I’m not completely sure. But I guess “alternative=less” is out of the way.

Reply:
Yes. That's why I went this way. Thought it was worth mentioning as something to consider. Let's make sure the reviewing maintainer is on the same page.

Reply:
Sounds sensible and there is a precedent too 👍